apps/web/components/apps/installation/ConfigureStepCard.tsx

```diff
@@ -116,82 +117,139 @@ const EventTypeAppSettingsForm = forwardRef<HTMLButtonElement, EventTypeAppSetti
   }
 );
 
-export const ConfigureStepCard: FC<ConfigureStepCardProps> = ({
-  loading,
-  formPortalRef,
-  eventTypes,
-  setConfigureStep,
-  handleSetUpLater,
+const EventTypeGroup = ({
+  groupIndex,
+  eventTypeGroups,
+  setUpdatedEventTypesStatus,
+  submitRefs,
   ...props
+}: ConfigureStepCardProps & {
+  groupIndex: number;
+  setUpdatedEventTypesStatus: Dispatch<SetStateAction<TUpdatedEventTypesStatus>>;
+  submitRefs: Array<React.RefObject<HTMLButtonElement>>;
 }) => {
-  const { t } = useLocale();
-  const { control, getValues } = useFormContext<TEventTypesForm>();
+  const { control } = useFormContext<TEventTypesForm>();
   const { fields, update } = useFieldArray({
     control,
-    name: "eventTypes",
+    name: `eventTypeGroups.${groupIndex}.eventTypes`,
+    keyName: "fieldId",
+  });
+  return (
+    <div className="ml-2 flex flex-col space-y-6">
+      {fields.map(
+        (field, index) =>
+          field.selected && (
+            <EventTypeAppSettingsForm
+              key={field.fieldId}
+              eventType={field}
+              loading={props.loading}
+              handleDelete={() => {
+                const eventTypeDb = eventTypeGroups[groupIndex].eventTypes?.find(
+                  (eventType) => eventType.id == field.id
+                );
+                update(index, {
+                  ...field,
+                  selected: false,
+                  metadata: eventTypeDb?.metadata,
+                  bookingFields: eventTypeDb?.bookingFields,
+                  ...(eventTypeDb?.locations && { locations: eventTypeDb.locations }),
+                });
```

Comment:
why do we need `eventTypeDb`  here? can we use 

```
update(index, { ...field, selected:false,
 metadata: field?.metadata, 
 bookingFields: field?.bookingFields })
```

Reply:
the reason I added eventTypeDb was to reset the form values to values from DB, fields would contain the updated values

use case - if we change something let's say price, now the form will have the new price "22" set, but when I remove this event by clicking on the cross on the top right corner we must reset the form to match the DB values, so that if we by any chance add it again we don't have the values prefilled in this case "22"
![Screenshot 2024-07-17 at 5 03 00 PM](https://github.com/user-attachments/assets/34840eda-30aa-461e-a6a5-3ea6f6ab32f0)